### DISCUSSION TITLE
Loki: Improve parser error when switching between code modes

### DISCUSSION THREAD

**Discussion by ivanahuckova** [grafana/grafana#56847] [Observability]
When switching between modes, the error message is confusing. Moreover, we have information about which part of the query we weren't able to parse, so we could improve that.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
